mitosheet/mitosheet/step_performers/merge.py

```diff
@@ -36,146 +37,70 @@ def step_type(cls) -> str:
 
     @classmethod
     def execute(cls, prev_state: State, params: Dict[str, Any]) -> Tuple[State, Optional[Dict[str, Any]]]:
-        how: str = get_param(params, 'how')
-        sheet_index_one: int = get_param(params, 'sheet_index_one')
-        sheet_index_two: int = get_param(params, 'sheet_index_two')
-        merge_key_column_ids: List[List[ColumnID]] = get_param(params, 'merge_key_column_ids')
-        selected_column_ids_one: List[ColumnID] = get_param(params, 'selected_column_ids_one')
-        selected_column_ids_two: List[ColumnID] = get_param(params, 'selected_column_ids_two')
-
-        merge_keys_one = prev_state.column_ids.get_column_headers_by_ids(sheet_index_one, list(map(lambda x: x[0], merge_key_column_ids)))
-        merge_keys_two = prev_state.column_ids.get_column_headers_by_ids(sheet_index_two, list(map(lambda x: x[1], merge_key_column_ids)))
-
-        selected_columns_one = prev_state.column_ids.get_column_headers_by_ids(sheet_index_one, selected_column_ids_one)
-        selected_columns_two = prev_state.column_ids.get_column_headers_by_ids(sheet_index_two, selected_column_ids_two)
-
-        # We create a shallow copy to make the new post state
-        post_state = prev_state.copy()
-
-        pandas_start_time = perf_counter()
-        new_df = _execute_merge(
-            prev_state.dfs,
-            prev_state.df_names,
-            how,
-            sheet_index_one,
-            merge_keys_one,
-            selected_columns_one,
-            sheet_index_two,
-            merge_keys_two,
-            selected_columns_two
-        )
-        pandas_processing_time = perf_counter() - pandas_start_time
-
-        # Add this dataframe to the new post state
-        post_state.add_df_to_state(new_df, DATAFRAME_SOURCE_MERGED)
-
-        return post_state, {
-            'pandas_processing_time': pandas_processing_time
+
+        new_df_name = get_first_unused_dataframe_name(prev_state.df_names, "df_merge")
+
+        execution_data = {
+            'new_df_name': new_df_name
         }
+        try:
+            return cls.execute_through_transpile(
+                prev_state, 
+                params, 
+                execution_data,
+                new_dataframe_params={
+                    'df_source': DATAFRAME_SOURCE_MERGED,
+                    'new_df_names': [new_df_name],
+                    'sheet_indexes': None
+                }
+            )
+
+        except ValueError:
```

Comment:
In this case moving the error handing until after we execute the code seems fine, in other cases we might want to do error handing first. ie: if we know the user is about to launch a super long running operation that we don't want to let them do. But we can still do that by moving it higher in the execute function so this seems fine